src/engine-scripts/puppet/jsReady.js

```diff
@@ -0,0 +1,31 @@
+module.exports = async (page) => {
+	await page.emulateMediaFeatures([
+		{ name: 'prefers-reduced-motion', value: 'reduce' },
+	]);
+	await page.evaluate( () => {
+		window.dispatchEvent( new Event( 'resize' ) );
+		const skin = mw.config.get( 'skin' );
+		let moduleName;
+		switch( skin ) {
+			case 'vector-2022':
+				moduleName = 'skins.vector.js';
+				break;
+			case 'vector':
+				moduleName = 'skins.vector.legacy.js';
+				break;
+			default:
+				return true;
+		}
+		return mw.loader.getState( moduleName ) === 'ready';
```

Comment:
Should this be a promise? e.g.

```
return mw.loader.using( moduleName )
```
As it stands, it doesn't look like its waiting for the module to finish loading. It's looks like its only checking it's current state?

Reply:
No using would have side effects. 

I think I'm getting confused by the evaluate method. I think a setInterval that resolves when this is true is perhaps what I need?

My understanding is evaluate has to resolve to true to avoid timeouts.

Reply:
The latest looks fine to me. I'm curious where you heard that it needs to resolve to true though? The puppeteer docs have examples that don't strictly return true e.g. https://pptr.dev/#?product=Puppeteer&version=v13.6.0&show=api-pageevaluatepagefunction-args

Reply:
When I was working on the scroll in the other patch, the scroll was synchronois and I was getting time outs if I didn't return something and didn't use an asynchronous function inside evaluate.  I have found the documentation a little frustrating as it hasn't matched up with what I've been seeing